dspace-server-webapp/src/test/java/org/dspace/app/rest/SystemWideAlertRestRepositoryIT.java

```diff
@@ -0,0 +1,299 @@
+
+import static com.jayway.jsonpath.JsonPath.read;
+import static com.jayway.jsonpath.matchers.JsonPathMatchers.hasJsonPath;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.startsWith;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.alerts.AllowSessionsEnum;
+import org.dspace.alerts.SystemWideAlert;
+import org.dspace.alerts.service.SystemWideAlertService;
+import org.dspace.app.rest.model.SystemWideAlertRest;
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.dspace.authorize.AuthorizeException;
+import org.junit.After;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Test class to test the operations in the SystemWideAlertRestRepository
+ */
+public class SystemWideAlertRestRepositoryIT extends AbstractControllerIntegrationTest {
+
+    private static final Logger log = LogManager.getLogger(SystemWideAlertRestRepositoryIT.class);
+
+    @Autowired
+    private SystemWideAlertService systemWideAlertService;
+
+    @After
+    public void destroy() throws Exception {
+        context.turnOffAuthorisationSystem();
+        systemWideAlertService.findAll(context).stream().forEach(systemWideAlert -> {
+            try {
+                systemWideAlertService.delete(context, systemWideAlert);
+            } catch (SQLException | IOException | AuthorizeException e) {
+                log.error(e);
+            }
+        });
+        context.restoreAuthSystemState();
+
+        super.destroy();
+    }
+
+
+    @Test
+    public void findAllTest() throws Exception {
+        Date countdownDate = new Date();
+        SystemWideAlert systemWideAlert1 = systemWideAlertService.create(context, "Test alert 1",
+                                                                         AllowSessionsEnum.ALLOW_CURRENT_SESSIONS_ONLY,
+                                                                         countdownDate,
+                                                                         true);
+        SystemWideAlert systemWideAlert2 = systemWideAlertService.create(context, "Test alert 2",
```

Comment:
Why do we have a test that allows for the creation of 2 alerts when we don't support that in the `SystemWideAlertRestRepository`?  (See `createSystemWideAlert()` method in that class)